### PR TITLE
8339870: Remove yet more stale TODOs

### PR DESCRIPTION
**Pull request by earthling-amzn**
Final round of TODO removal.
* Move asterisk for pointer next to type, not variable
* Remembered set verifier uses a small 'scanner' abstraction to simplify reading of read or write card table

## Testing

GHA, Internal pipelines

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8339870](https://bugs.openjdk.org/browse/JDK-8339870): Remove yet more stale TODOs (**Task** - P4)

### Reviewers
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Committer) ⚠️ Review applies to [8719ba9d](https://git.openjdk.org/shenandoah/pull/496/files/8719ba9d0f670ba2b4693091dff5733ca1860b9e)
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer) ⚠️ Review applies to [8719ba9d](https://git.openjdk.org/shenandoah/pull/496/files/8719ba9d0f670ba2b4693091dff5733ca1860b9e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/496/head:pull/496` \
`$ git checkout pull/496`

Update a local copy of the PR: \
`$ git checkout pull/496` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/496/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 496`

View PR using the GUI difftool: \
`$ git pr show -t 496`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/496.diff">https://git.openjdk.org/shenandoah/pull/496.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/496#issuecomment-2341532851)

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp [hunk 1]
```diff
@@ -82,19 +82,25 @@ void ShenandoahDirectCardMarkRememberedSet::mark_range_as_clean(size_t card_inde
...
 
-bool ShenandoahDirectCardMarkRememberedSet::is_card_dirty(HeapWord *p) const {
+bool ShenandoahDirectCardMarkRememberedSet::is_card_dirty(HeapWord* p) const {
   size_t index = card_index_for_addr(p);
   CardValue* bp = &(_card_table->read_byte_map())[index];
   return (bp[0] == CardTable::dirty_card_val());
 }
 
-void ShenandoahDirectCardMarkRememberedSet::mark_card_as_dirty(HeapWord *p) {
+bool ShenandoahDirectCardMarkRememberedSet::is_write_card_dirty(HeapWord* p) const {
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp:91]: Interesting. Is this new? Is it used somewhere or introduced for the purpose of a more consistent/uniform API?

(Not requiring any change; just asking. I am all for uniform/consistent APIs.)

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp [hunk 2]
```diff
@@ -307,35 +313,39 @@ bool ShenandoahScanRemembered::is_write_card_dirty(size_t card_index) {
...
   return _rs->is_card_dirty(p);
 }
 
-void ShenandoahScanRemembered::mark_card_as_dirty(HeapWord *p) {
+void ShenandoahScanRemembered::mark_card_as_dirty(HeapWord* p) {
   _rs->mark_card_as_dirty(p);
 }
 
-void ShenandoahScanRemembered::mark_range_as_dirty(HeapWord *p, size_t num_heap_words) {
+bool ShenandoahScanRemembered::is_write_card_dirty(HeapWord* p) {
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp:324]: Same question/remark as I asked/made for the ShenDirectCardMarkRS api above at line 91.

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp [hunk 3]
```diff
@@ -878,7 +866,7 @@ class ShenandoahScanRemembered: public CHeapObj<mtGC> {
 // A ShenandoahRegionChunk represents a contiguous interval of a ShenandoahHeapRegion, typically representing
 // work to be done by a worker thread.
 struct ShenandoahRegionChunk {
-  ShenandoahHeapRegion *_r;      // The region of which this represents a chunk
+  ShenandoahHeapRegion* _r;      // The region of which this represents a chunk
   size_t _chunk_offset;          // HeapWordSize offset
   size_t _chunk_size;            // HeapWordSize qty
 };
```
Comment (ysramakrishna) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp:872]: At line 880 below:

"other worker will threads repeatedly ..." should be "other worker threads will repeatedly".

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp (hunk 1 above)
Comment (ysramakrishna) [reply src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp:91]: Never mind. Found the (new) place where it's now used.

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp (hunk 2 above)
Comment (ysramakrishna) [reply src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp:324]: Never mind; found where it's used now.

**Review by ysramakrishna — APPROVED** [review]
Looks good. Thanks for the cleanups!

**Review by kdnilsen — APPROVED** [review]

File: src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp [hunk 4]
```diff
@@ -682,13 +682,6 @@ void ShenandoahConcurrentGC::op_final_mark() {
...
-    //
-    // TODO: Under severe memory overload conditions that can be checked here, we may want to limit
-    // the inclusion of old-gen candidates within the collection set.  This would allow us to prioritize efforts on
-    // evacuating young-gen,  This remediation is most appropriate when old-gen availability is very high (so there
-    // are negligible negative impacts from delaying completion of old-gen evacuation) and when young-gen collections
-    // are "under duress" (as signalled by very low availability of memory within young-gen, indicating that/ young-gen
-    // collections are not triggering frequently enough).
     _generation->prepare_regions_and_collection_set(true /*concurrent*/);
 
     // Upon return from prepare_regions_and_collection_set(), certain parameters have been established to govern the
```
[the remark this answers is not in the document]
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp:687]: Removed the block of comments and improved the comment on promotion reserve in old generation.

File: src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp [hunk 5]
```diff
@@ -729,11 +729,6 @@ void ShenandoahFullGC::distribute_slices(ShenandoahHeapRegionSet** worker_slices
 #endif
 }
 
-// TODO:
-//  Consider compacting old-gen objects toward the high end of memory and young-gen objects towards the low-end
-//  of memory.  As currently implemented, all regions are compacted toward the low-end of memory.  This creates more
-//  fragmentation of the heap, because old-gen regions get scattered among low-address regions such that it becomes
-//  more difficult to find contiguous regions for humongous objects.
```
[the remark this answers is not in the document]
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp:736]: Yes, have this in an internal ticket (for now).

**Pull request closed without merging**